src/chains/ethereum/ethereum/src/forking/trie.ts

```diff
@@ -166,16 +179,21 @@ export class ForkTrie extends GanacheTrie {
 
       const hash = keccak(key);
       const { node, stack } = await this.findPath(hash);
-      if (node) await this._deleteNode(hash, stack);
+      if (node) {
+        await this._deleteNode(hash, stack);
+        await this.persistRoot();
+      }
 
       await metaDataPutPromise;
     } else {
       const hash = keccak(key);
       const { node, stack } = await this.findPath(hash);
-      if (node) await this._deleteNode(hash, stack);
+      if (node) {
+        await this._deleteNode(hash, stack);
+        await this.persistRoot();
```

Comment:
Done ish. I added a note for the inspiration for this function. Some digging would be needed for why `persistRoot` needs to be called, the honest answer is because that's what they do!